go/networkqualityd.go

```diff
@@ -91,17 +100,63 @@ func main() {
 		}()
 	}
 
+	if *announce {
+		ips := make([]net.IP, 0)
+		// The user may give us a hostname (rather than an address to listen on). In order to
+		// handle this situation, we will use DNS to convert it to an IP. As a result, we may
+		// get back more than one address -- handle that!
+		if addresses, lookupErr := net.LookupHost(*listenAddr); lookupErr == nil {
+			for _, addr := range addresses {
+				if parsedAddr := net.ParseIP(addr); parsedAddr != nil {
+					ips = append(ips, parsedAddr)
+				}
+			}
+		}
+		if announceResponder, announceHandle, announceErr := configureAnnouncer(ips, *configName, *configPort); announceErr == nil {
+			defer announceResponder.Remove(announceHandle)
+			go announceResponder.Respond(operatingCtx)
+		} else {
+			log.Printf("Warning: Could not announce the server instance: %v.\n", announceErr)
+		}
+	}
+
 	var wg sync.WaitGroup
 	wg.Add(1)
 
-	listenAddr := fmt.Sprintf("%s:%d", *listenAddr, *configPort)
-	go func(listenAddr string) {
+	server := &http.Server{}
+	server.Addr = fmt.Sprintf("%s:%d", *listenAddr, *configPort)
+	server.Handler = mux
+
+	// Create an http2.Server simply for the purposes of configuring
```

Comment:
Was this section intended for this PR? This looks like the stuff from now closed PR #19

Reply:
It is/was. But, we will still need access to a `server` because we are going to cancel it down below. I will refactor to make it cleaner. Good catch!